Rubberduck.RibbonDispatcher/Sample Ribbons/RibbonLibrary/src/RibbonCallbacksAdapter.bas

```diff
@@ -0,0 +1,226 @@
+Attribute VB_Name = "RibbonCallbacksAdapter"
+Option Explicit
+
+Public Const ModuleName   As String = "RibbonCallbacksAdapter."
+Public Const DefaultImage As String = "MacroSecurity"
+
+Public Enum ControlSize
+    rdRegular = RdControlSize_rdRegular
+    rdLarge = RdControlSize_rdLarge
+End Enum
+
+Private Function ViewModelFor(ByVal Control As IRibbonControl _
+) As Rubberduck_RibbonDispatcher.RibbonViewModel
+    On Error GoTo EH
+    Dim WkBk As IRibbonWorkbook
+    Set WkBk = Control.Context.Parent
+    Set ViewModelFor = WkBk.ViewModel
+XT: Exit Function
+EH: If Err.Number <> 438 Then ReraiseError Err, ModuleName & "NewRibbonModel"
```

Comment:
Please keep consistent with error handlers of unit test Rubberduck-templated/generated VBA code; they have `TestExit` and `TestFail`, without any instructions on the label line. So:

    Private Function ViewModelFor(ByVal Control As IRibbonControl) As Rubberduck_RibbonDispatcher.RibbonViewModel
        On Error GoTo CleanFail
        '...

    CleanExit:
        Exit Function
    CleanFail:
        With Err
            If .Number <> 438 Then .Raise .Number
        End With
        Resume CleanExit
    End Function

Reply:
I have been waiting for someone to look closely enough at the code to provide guidance on this. 

How does the Rubberduck team generate a stack trace from VBA? Don't you find that a time-killer on problem diagnosis.

Reply:
Might be I don't write that many bugs, but IMO generating a stack trace in VBA is bug-prone, time-consuming and eventually outright misleading, being stringly-typed and thus not refactor-friendly. "Break on all errors" with well-placed breakpoints and no global state has served me quite well.

Reply:
You are right - it is ***a hard problem***, and error prone. You should try tackling more of those, since you are such a perfect coder. While you are at that, perhaps think a bit on the difference between *easily scanned code* and *easily read code*. Any high school student these days can write code to a style guide that the compiler understands - writing code that a human can actually understand; code that steers the eyes towards important or difficult areas, and provides quick reassurance that others are *as they should be*, is what a professional programmer provides.

Reply:
With he modern IDE, I could resolve every single violation of your style preferences in an hour or so. Change a couple of my formatting options, then proceed with the following key-chords through every module: Ctrl-A; Ctrl-X; Ctrl-V. Has it ever occurred to you that perhaps I have tackled subtle and difficult problems? That there are areas of the code that look trivial but that aren't? Areas where perhaps a better solution would be apparent to another pair of eyes, or after an other release of Visual Studio. That a determined effort has been made to try and make the code readable rather than merely scannable. You don't even ask for my reasons for when I have violated the style guidelines. 

It is a fundamental tenet of style that, generally, conformity must be made. Only by generally adhering to the style, can one make important change in emphasis by judicial violations. I strive to achieve this. Perhaps I have missed the mark by too frequent violations. Perhaps my use of this technique is occasionally too subtle. I am open to suggestions for better ways to make this communication, but your broad-brush insistence on a three-colour finger-paint palette is simplistic and inadequate. Code is too complex and difficult to be accurately described with finger paints.

Reply:
Consider the Mixin classes. I (also) always suffix my static Extension classes with "Extensions"; except here. Why? Because these are NOT EXTENSION classes. No other random methods are welcome in these classes, so don't go throwing them in. So when some future maintainer goes looking for an Extensions class to throw a new method into, just maybe (s)he will think about the style rule, and realize that non of these are appropriate places for such. Every time that happens, and the method gets appropriately added elsewhere, is a double victory - for the style rule AND for the (appropriate) violation.